Monika After Story/game/definitions.rpy

```diff
@@ -3743,6 +3744,37 @@ init -991 python in mas_utils:
     mas_log.write("VERSION: {0}\n".format(store.persistent.version_number))
 
 
+    def weightedChoice(choice_weight_tuple_list):
+        """
+        Returns a random item based on weighting.
+        NOTE: That weight essentially corresponds to the equivalent of how many times to duplicate the choice
+
+        IN:
+            choice_weight_tuple_list - List of tuples with the form (choice, weighting)
+
+        OUT:
+            random choice value picked using choice weights
+        """
+        #No items? Just return None
+        if not choice_weight_tuple_list:
+            return None
+
+        choices, weights = zip(*choice_weight_tuple_list)
+        total_weight = 0
+        cumulative_weights = list()
+
+        #Now we collect all the weights and geneate a cumulative and total weight amount
+        for weight in weights:
+            total_weight += weight
+            cumulative_weights.append(total_weight)
+
+        #Now bisect the cumulative weight using a random point
+        r_index = bisect(
```

Comment:
from the python docs on `bisect`: `This module provides support for maintaining a list in sorted order without having to sort the list after each insertion`. Based on this documentation, your use of `bisect` seems incorrect.

But after scrolling down the page, I see what you are trying to do here (numeric table lookups based on a set of breakpoints - weights). Can you add a more descriptive comment(s) here on what `bisect` is doing for you in this context? aka something like "bisect finds the index of closest weight to the random weight that we select, which is then used to return the appropriate choice"?

Additionally, doesn't `cumulative_weights` (and the choices array) need to be sorted for this to work right? `bisect` doesn't do sorting, it just acts on an already sorted list. If you need sorting, there is the `insort` functions in `bisect`, otherwise I have generalized insertion sort functions in `filtering-backgrounds`, I could move them out if needed.